collectors/plugins.d/pluginsd_parser.c

```diff
@@ -561,11 +578,22 @@ PARSER_RC pluginsd_overwrite(char **words, void *user, PLUGINSD_ACTION  *plugins
     return PARSER_RC_OK;
 }
 
+
+#ifdef ENABLE_DBENGINE
 PARSER_RC pluginsd_guid(char **words, void *user, PLUGINSD_ACTION *plugins_action)
 {
     char *uuid_str = words[1];
     uuid_t uuid;
 
+    RRDHOST *host = ((PARSER_USER_OBJECT *) user)->host;
+
+    if (host->rrd_memory_mode != RRD_MEMORY_MODE_DBENGINE) {
+        return PARSER_RC_OK;
+    }
+
+    if (((PARSER_USER_OBJECT *) user)->cd->version < VERSION_GUID)
```

Comment:
I don't like how we have separate versions for streaming and the plugins.D language itself, but in practice we use the streaming version interchangeably in such a way that influences the language parsing itself.

At some point we should clearly define separate language version and streaming protocol versions.

Reply:
I don't see any need to use the streaming version for anything other than defining the version of the streaming protocol.